services/server/src/handlers/user.ts

```diff
@@ -189,122 +188,116 @@ export async function create(req: Request, res: Response, next: NextFunction) {
   res.status(200).json(response);
 }
 
-function update(_: ApiContext) {
-  return safeRequest(
-    async (req: Request, res: Response, next: NextFunction) => {
-      const { id } = identityObject.parse(req.params);
-
-      const currentUser = req.user;
-      const allowed = isUser(currentUser);
-      if (!allowed) return next(forbidden());
-
-      const targetUser = await users.findById(id);
-      if (!targetUser) return next(notfound.user());
-
-      const targetTutor = await tutors.findById(targetUser.id);
-
-      const {
-        email,
-        name,
-        password,
-        gender,
-        birthYear,
-        image,
-        thumbnail,
-        video,
-        bio,
-        about,
-        notice,
-        phone,
-        city,
-      }: IUser.UpdateApiPayload = updateUserPayload.parse(req.body);
-
-      // return forbidden if the currentUser is neither admin nor studio and tring to update other user data
-      if (currentUser.id !== targetUser.id && isRegularUser(currentUser))
-        return next(forbidden());
-
-      // return forbidden if the studio is trying to drop user image
-      if (isStudio(currentUser) && image === null) return next(forbidden());
-
-      // return forbidden if the studio is trying to drop media for an unassociated user
-      if (isStudio(currentUser) && targetTutor?.studioId !== currentUser.id)
-        return next(forbidden());
-
-      if (password) {
-        const expectedPasswordHash = await users.findUserPasswordHash(
-          targetUser.id
-        );
+export async function update(req: Request, res: Response, next: NextFunction) {
+  const { id } = identityObject.parse(req.params);
+
+  const currentUser = req.user;
+  const allowed = isUser(currentUser);
+  if (!allowed) return next(forbidden());
+
+  const targetUser = await users.findById(id);
+  if (!targetUser) return next(notfound.user());
+
+  const targetTutor = await tutors.findById(targetUser.id);
+
+  const {
+    email,
+    name,
+    password,
+    gender,
+    birthYear,
+    image,
+    thumbnail,
+    video,
+    bio,
+    about,
+    notice,
+    phone,
+    city,
+  }: IUser.UpdateApiPayload = updateUserPayload.parse(req.body);
+
+  // return forbidden if the currentUser is neither admin nor studio and tring to update other user data
+  if (currentUser.id !== targetUser.id && isRegularUser(currentUser))
+    return next(forbidden());
```

Comment:
This is an important check! 

----

* Instead of describing the code, let's describe the "rules". 
* Let's make the code self descriptive.

```suggestion
  // It is forbidden for regular users to update other users.
  const updatingSelf = currentUser.id === targetUser.id
  if (!updatingSelf && isRegularUser(currentUser)) return next(forbidden());
```